algorithms/kernel/pca/oneapi/pca_dense_correlation_batch_kernel_ucapi_impl.i

```diff
@@ -233,6 +234,106 @@ services::Status PCACorrelationKernelUCAPI<algorithmFPType>::calculateVariances(
+
+    BlockDescriptor<algorithmFPType> corrBlock;
+    correlation.getBlockOfRows(0, nFeatures * nFeatures, readOnly, corrBlock);
+    services::Buffer<algorithmFPType> correlationBuffer = corrBlock.getBuffer();
+
+    auto fullEigenvectors = context.allocate(TypeIds::id<algorithmFPType>(), nFeatures * nFeatures, &status);
+    DAAL_CHECK_STATUS_VAR(status);
+    context.copy(fullEigenvectors, 0, correlationBuffer, 0, nFeatures * nFeatures, &status);
+    DAAL_CHECK_STATUS_VAR(status);
+
+    auto fullEigenvalues = context.allocate(TypeIds::id<algorithmFPType>(), nFeatures, &status);
+    DAAL_CHECK_STATUS_VAR(status);
+
+    status |= computeEigenvectorsInplace(context, nFeatures, fullEigenvectors, fullEigenvalues);
+    DAAL_CHECK_STATUS_VAR(status);
+    status |= sortEigenvectorsDescending(context, nFeatures, fullEigenvectors, fullEigenvalues);
+    DAAL_CHECK_STATUS_VAR(status);
+
+    {
+        BlockDescriptor<algorithmFPType> eigenvectorsBlock;
+        eigenvectors.getBlockOfRows(0, nFeatures * nComponents, writeOnly, eigenvectorsBlock);
+        services::Buffer<algorithmFPType> eigenvectorsBuffer = eigenvectorsBlock.getBuffer();
+
+        context.copy(eigenvectorsBuffer, 0, fullEigenvectors, 0, nFeatures * nComponents, &status);
+        DAAL_CHECK_STATUS_VAR(status);
+
+        status |= eigenvectors.releaseBlockOfRows(eigenvectorsBlock);
+        DAAL_CHECK_STATUS_VAR(status);
+    }
+
+    {
+        BlockDescriptor<algorithmFPType> eigenvaluesBlock;
+        eigenvalues.getBlockOfRows(0, nComponents, writeOnly, eigenvaluesBlock);
+        services::Buffer<algorithmFPType> eigenvaluesBuffer = eigenvaluesBlock.getBuffer();
+
+        context.copy(eigenvaluesBuffer, 0, fullEigenvalues, 0, nComponents, &status);
+        DAAL_CHECK_STATUS_VAR(status);
+
+        status |= eigenvalues.releaseBlockOfRows(eigenvaluesBlock);
+        DAAL_CHECK_STATUS_VAR(status);
+    }
+
+    status |= correlation.releaseBlockOfRows(corrBlock);
+    DAAL_CHECK_STATUS_VAR(status);
+
+    return status;
+}
+
+template <typename algorithmFPType>
+services::Status PCACorrelationKernelUCAPI<algorithmFPType>::computeEigenvectorsInplace(ExecutionContextIface & context, const size_t nFeatures,
+                                                                                        UniversalBuffer & fullEigenvectors,
+                                                                                        UniversalBuffer & fullEigenvalues)
+{
+    services::Status status;
+
+    const math::Job jobz  = math::Job::vec;
+    const math::UpLo uplo = math::UpLo::Upper;
+
+    const size_t lwork  = 2 * nFeatures * nFeatures + 6 * nFeatures + 1;
+    const size_t liwork = 5 * nFeatures + 3;
+
+    auto work = context.allocate(TypeIds::id<algorithmFPType>(), lwork, &status);
+    DAAL_CHECK_STATUS_VAR(status);
+    auto iwork = context.allocate(TypeIds::id<int64_t>(), liwork, &status);
+    DAAL_CHECK_STATUS_VAR(status);
```

Comment:
What the purpose of this auxiliary staff?

Reply:
align to CPU

Reply:
Can you explain to me what `work`, `lwork`, `iwork` and `liwork` are used for?